code/datums/elements/immerse.dm

```diff
@@ -227,7 +229,7 @@ GLOBAL_LIST_INIT(immerse_ignored_movable, typecacheof(list(
  */
 /datum/element/immerse/proc/try_immerse(atom/movable/movable, atom/movable/buckled)
 	var/atom/movable/to_check = buckled || movable
-	if(!(to_check.movement_type & MOVETYPES_NOT_TOUCHING_GROUND) && !movable.throwing)
+	if(!(to_check.movement_type & MOVETYPES_NOT_TOUCHING_GROUND) && !movable.throwing && to_check.has_gravity())
```

Comment:
I think this `has_gravity` is redundant?

People without gravity are floating which is already covered by the movetype check